src/detect.c

```diff
@@ -804,8 +804,14 @@ static inline void DetectRulePacketRules(
 
         if (!(sflags & SIG_FLAG_NOALERT)) {
             /* stateful sigs call PacketAlertAppend from DeStateDetectStartDetection */
-            if (!state_alert)
-                PacketAlertAppend(det_ctx, s, p, 0, alert_flags);
+            if (!state_alert) {
```

Comment:
if the alert doesn't happen in a tx, which is the case here, we can't know the tx id for sure

Reply:
Do you mean that, in this case, we should not have any app-layer metadata as they come from a transaction ?

Reply:
Essentially yes. I guess we could see if there is a single tx and use that as a special case. But if there are more txs we'd risk logging the wrong tx causing confusion.

Reply:
Thanks, closing this for now, and moving to the redmine ticket for further discussion